Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,332 @@
+  index = defaultValue(index, 0);
+  return this._colorTextures[index];
+};
+
+FramebufferManager.prototype.setColorTexture = function (texture, index) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createColorAttachments) {
+    throw new DeveloperError(
+      "createColorAttachments must be false if setColorTexture is called."
+    );
+  }
+  if (index >= this._colorAttachmentsLength) {
+    throw new DeveloperError(
+      "index must be smaller than total number of color attachments."
+    );
+  }
+  //>>includeEnd('debug');
+  index = defaultValue(index, 0);
+  this._colorTextures[index] = texture;
+};
+
+FramebufferManager.prototype.getDepthRenderbuffer = function () {
+  return this._depthRenderbuffer;
+};
+
+FramebufferManager.prototype.setDepthRenderbuffer = function (renderbuffer) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthRenderbuffer is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthRenderbuffer = renderbuffer;
+};
+
+FramebufferManager.prototype.getDepthTexture = function () {
+  return this._depthTexture;
+};
+
+FramebufferManager.prototype.setDepthTexture = function (texture) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthTexture is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthTexture = texture;
+};
+
+FramebufferManager.prototype.getDepthStencilRenderbuffer = function () {
+  return this._depthStencilRenderbuffer;
+};
+
+FramebufferManager.prototype.setDepthStencilRenderbuffer = function (
+  renderbuffer
+) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthStencilRenderbuffer is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthStencilRenderbuffer = renderbuffer;
+};
+
+FramebufferManager.prototype.getDepthStencilTexture = function () {
+  return this._depthStencilTexture;
+};
+
+FramebufferManager.prototype.setDepthStencilTexture = function (texture) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthStencilTexture is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthStencilTexture = texture;
+};
+
+FramebufferManager.prototype.clear = function (
+  context,
+  passState,
+  clearColor,
+  clearCommand
+) {
+  var framebuffer = passState.framebuffer;
+  var clearCommandColor = clearCommand.color;
+
+  passState.framebuffer = this._framebuffer;
+  clearCommand.color = Color.clone(clearColor, clearCommand.color);
+  clearCommand.execute(context, passState);
+
+  clearCommand.color = clearCommandColor;
+  passState.framebuffer = framebuffer;
```

Comment:
Is it ever necessary to temporarily set the pass state's framebuffer instead of a command's framebuffer? I'm wondering if the `passState` argument can be removed, since `Context` has a default pass state to use if one isn't given.

Reply:
> Is it ever necessary to temporarily set the pass state's framebuffer instead of a command's framebuffer?

I don't think so. But the pass state argument should still passed in because its other properties might be important.